pallets/utxo/rpc/runtime-api/src/lib.rs

```diff
@@ -25,6 +26,8 @@ sp_api::decl_runtime_apis! {
         // At the moment we have some problems with use serde in RPC, we can serialize and deserialize
         // only simple types. This approach allow us to return Vec<(TokenId, TokenName)> instead of
         // pallet_utxo_tokens::TokenListData
-        fn tokens_list() -> Vec<(u64, Vec<u8>)>;
+        fn tokens_list() -> Vec<(H256, Vec<u8>)>;
+        // Getting NFT data
+        fn nft_read(id: H256) -> Option<(/* Data url */ Vec<u8>, /* Data hash */ [u8; 32])>;
```

Comment:
I think the hash shouldn't be a `[u8, 32]`, but some `impl` to something that provides a generic hash interface. It's just nicer. Not a big deal.

Reply:
Also, there can be a data type that "represents" an NFT, which will have both of these. I'm not a big fan of using tuples on arguments as they can be confusing. Instead of having a tuple commented somewhere hidden (like this) in the code, perhaps it's a better idea to just make a type that represents an NFT. That type will have tall the impls that will deal with hashes, initialization, checks, etc.

Reply:
Sure

Reply:
Okay, there is a problem:

If we change a tuple to a struct in `pallets/utxo/RPC/src/lib.rs:35`

```rust 
#[rpc]
pub trait UtxoApi<BlockHash> {
...
    #[rpc(name = "tokens_list")]
    fn tokens_list(&self, at: Option<BlockHash>) -> Result<Vec<TokenInstance>>;

```

We had this error:
```bash 
error[E0277]: the trait bound `pallet_utxo_tokens::TokenInstance: Deserialize<'_>` is not satisfied
  --> pallets/utxo/rpc/src/lib.rs:28:1
   |
28 | #[rpc]
   | ^^^^^^ the trait `Deserialize<'_>` is not implemented for `pallet_utxo_tokens::TokenInstance`
   |
   = note: required because of the requirements on the impl of `for<'de> Deserialize<'de>` for `Vec<pallet_utxo_tokens::TokenInstance>`
   = note: required because of the requirements on the impl of `DeserializeOwned` for `Vec<pallet_utxo_tokens::TokenInstance>`
   = note: this error originates in the attribute macro `rpc` (in Nightly builds, run with -Z macro-backtrace for more info)

```
Obviously, I need to make an implementation: 
```
 impl of `for<'de> Deserialize<'de>` for `Vec<pallet_utxo_tokens::TokenInstance>`
```
But `jsonrpc_derive` has a `feature = ["std"]` but the UTXO panel doesn't, when I tried to add the std feature to the UTXO panel I had a bunch of errors.

Any ideas guys? @altonen @iljakuklic

Reply:
```bash
     Compiling pp-api v3.0.0 (/home/anton/mintlayer/core/traits/pp-api)
  error[E0432]: unresolved import `sp_core::to_substrate_wasm_fn_return_value`
    --> /home/anton/.cargo/git/checkouts/substrate-7e08433d4c370a21/6e15de9/primitives/api/src/lib.rs:80:9
     |
  80 | pub use sp_core::to_substrate_wasm_fn_return_value;
     |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `to_substrate_wasm_fn_return_value` in the root

  For more information about this error, try `rustc --explain E0432`.
```
Weird

Reply:
I don't know what is the proper Substrate way of doing things but looking at Parity's implementation, they don't use `jsonrpc_derive` for pallets so that's a bad sign. I also tried using JSON for a pallet call but it didn't go so well because of the `std` dependency. Are you able to use some other encoding?

Reply:
@altonen According to advice @TheQuantumPhysicist and Ben's decision we will remove RPC calls.